test/hotspot/jtreg/compiler/lib/generators/Generators.java

```diff
@@ -0,0 +1,573 @@
+ * should be tested with different distributions. For example, a function <code>h(int, int, int)</code> under test might
+ * be worthwhile to test not only with uniformly sampled integers but might also exhibit interesting behavior if tested
+ * specifically with powers of two. Suppose further that each argument has a different range of allowed values. We
+ * can write a test function as below:
+ *
+ * <pre><code>
+ * void test(Generator{@literal <Integer>} g) {
+ *     h(g.restricted(1, 10).next(), g.next(), g.restricted(-10, 100).next());
+ * }
+ * </code></pre>
+ *
+ * Then <code>test</code> can be called with different distributions, for example:
+ *
+ * <pre><code>
+ * test(G.uniformInts());
+ * test(G.specialInts(0));
+ * </code></pre>
+ * <p>
+ * If there is a single value that is interesting as an argument to all three parameters, we might even call this
+ * method with a single generator, ensuring that the single value is within the restriction ranges:
+ *
+ * <pre><code>
+ * test(G.single(1));
+ * </code></pre>
+ *
+ * <p>
+ * Furthermore, this class offers utility generators, such as {@link #randomElement(Collection)} or
+ * {@link #orderedRandomElement(Collection)} for sampling from a list of elements; {@link #single(Object)} for a
+ * generator that only produces a single value; and {@link #mixed(Generator, Generator, int, int)} which combines
+ * two generators with the provided weights.
+ * <p>
+ * Thus, the generators provided by this class are composable and therefore extensible. This allows to easily
+ * create random generators even with types and distributions that are not predefined. For example, to create a
+ * generator that provides true with 60 percent probably and false with 40 percent probably, one can simply write:
+ * <pre><code>G.mixed(G.single(true), G.single(false), 60, 40)</code></pre>
+ * <p>
+ * Generators are also by no means limited to work with numbers. Restrictable generators can work with any type that
+ * implements {@link Comparable} while generators such as {@link #randomElement(Collection)} and {@link #single(Object)}
+ * work with any type. Note that there are separate restrictable versions of the last two generators
+ * (namely, {@link #orderedRandomElement(Collection)} and {@link #single(Comparable)}) that work with comparable types.
+ * For example, you might restrict a generator choosing strings at random:
+ * <pre><code>G.orderedRandomElement(List.of("Bob", "Alice", "Carol")).restricted("Al", "Bz")</code></pre>
+ * <p>
+ * For all the generators created by instances of this class, the following rule applies: Integral generators are
+ * always inclusive of both the lower and upper bound, while floating point generators are always inclusive of the
+ * lower bound but always exclusive of the upper bound. This also applies to all generators obtained by restricting
+ * these generators further.
```

Comment:
It would be nice if you said that one is generally supposed to take a random distribution for tests where the distribution does not matter. So `G.ints()`. This is best practice, because that allows the test to be run under different conditions - maybe only a single distribution an trigger a bug.